src/module.c

```diff
@@ -873,6 +873,7 @@ void moduleCallCommandUnblockedHandler(client *c) {
  * because we only need to track of nesting level in the main thread
  * (only the main thread uses propagatePendingCommands) */
 void moduleCreateContext(RedisModuleCtx *out_ctx, RedisModule *module, int ctx_flags) {
+    serverAssert(out_ctx != NULL);
```

Comment:
Build: `make CC=gcc-12`
```sh
In function ‘memset’,
    inlined from ‘moduleCreateContext’ at module.c:877:5,
    inlined from ‘RM_GetDetachedThreadSafeContext’ at module.c:8410:5:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:59:10: warning: ‘__builtin_memset’ writing 104 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
   59 |   return __builtin___memset_chk (__dest, __ch, __len,
```

Reply:
Here is what I can't understand, why GCC saw out_ctx as a null ptr, I am not sure if this problem is related to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109175.